### PR TITLE
Document installation of gems on puppetserver

### PR DESCRIPTION
**Pull request by ghoneycutt**

### CODE REVIEW

**Comment by ghoneycutt** [comment]
This should be validated by someone with Puppet 5.5.8 / 6 and not just merged until then.

@threepistons could you have a look?

**Comment by threepistons** [comment]
> This should be validated by someone with Puppet 5.5.8 / 6 and not just merged until then.
> 
> @threepistons could you have a look?

```
root@obfuscated Tue Apr 16 10:22:47am 
/etc/yum.repos.d/ puppet --version  
5.5.12
root@obfuscated Tue Apr 16 10:27:39am 
/etc/yum.repos.d/ /opt/puppetlabs/server/bin/puppetserver gem install hiera-eyaml-gpg ruby_gpg
Fetching: optimist-3.0.0.gem (100%)
Successfully installed optimist-3.0.0
Fetching: highline-1.6.21.gem (100%)
Successfully installed highline-1.6.21
Fetching: hiera-eyaml-3.0.0.gem (100%)
Successfully installed hiera-eyaml-3.0.0
Fetching: hiera-eyaml-gpg-0.6.gem (100%)
Successfully installed hiera-eyaml-gpg-0.6
Fetching: ruby_gpg-0.3.2.gem (100%)
Successfully installed ruby_gpg-0.3.2
5 gems installed
root@obfuscated Tue Apr 16 10:29:32am 
/etc/yum.repos.d/ /opt/puppetlabs/puppet/bin/gem install hiera-eyaml-gpg ruby_gpg
Fetching: hiera-eyaml-gpg-0.6.gem (100%)
Successfully installed hiera-eyaml-gpg-0.6
Parsing documentation for hiera-eyaml-gpg-0.6
Installing ri documentation for hiera-eyaml-gpg-0.6
Done installing documentation for hiera-eyaml-gpg after 0 seconds
Fetching: ruby_gpg-0.3.2.gem (100%)
Successfully installed ruby_gpg-0.3.2
Parsing documentation for ruby_gpg-0.3.2
Installing ri documentation for ruby_gpg-0.3.2
Done installing documentation for ruby_gpg after 0 seconds
2 gems installed
```

**Comment by threepistons** [comment]
Hopefully it isn't important that I ran the `puppetserver gem` command before the `gem` command.

```
root@obfuscated Tue Apr 16 10:29:50am 
/etc/yum.repos.d/ puppetserver --version
puppetserver version: 6.3.0
```

**Comment by ghoneycutt** [comment]
@threepistons you'll need to restart the puppetserver after that. Did it work for you?

**Comment by threepistons** [comment]
Getting to decryption is my current WIP.

**Comment by threepistons** [comment]
I have a working hiera-eyaml-gpg at last!

`~/ puppetserver --version
puppetserver version: 6.3.0
`

GPG encrypted values go in, cleartext values come out.  The one problem I found is that feeding GPGed passwords into augeasproviders_grub's grub_user type via Hiera, the **cleartext** passwords appear in the agent output.  Luckily, grub_user accepts pbkdf2 hashed passwords.

**Comment by alexjfisher** [comment]
> The one problem I found is that feeding GPGed passwords into augeasproviders_grub's grub_user type via Hiera, the **cleartext** passwords appear in the agent output. Luckily, grub_user accepts pbkdf2 hashed passwords.

It's a bit off-topic, but you can probably also solve this (and similar issues) using the `Sensitive` data type.  Something a bit like... (untested code follows)

```puppet
class profile::grub
(
  Sensitive[String[1]] $password,
)
{
  grub_user { 'my_user':
    password => $password,
  }
}
```

```yaml
---
lookup_options:
  '^profile::grub::password:
    convert_to: 'Sensitive'
```

See https://puppet.com/blog/my-journey-securing-sensitive-data-puppet-code for some more details.

File: README.md [hunk 1]
```diff
@@ -45,6 +45,16 @@ should probably acquaint yourself with at https://github.com/TomPoulton/hiera-ey
 Note that in order to install the gpgme gem you'll need to have the ruby development package installed
 for your distribution.
 
+For use on puppetserver
+---
+
+As root
+
+```sh
+/opt/puppetlabs/puppet/bin/gem install hiera-eyaml-gpg ruby_gpg
```
Comment (alexjfisher) [inline README.md]: This is optional??  Perhaps just needed for `puppet lookup`?

Comment (threepistons) [reply README.md]: I would suspect so.  I did `/opt/puppetlabs/puppet/bin/gem uninstall hiera-eyaml-gpg ruby_gpg` in my Puppet master and it still merrily decrypts my values.

**Pull request merged**